net/http/authn/authn.go

```diff
@@ -60,6 +66,13 @@ func (a *API) Authenticate(req *http.Request) (*http.Request, error) {
 	return req.WithContext(ctx), nil
 }
 
+func (a *API) certAuthn(req *http.Request) (*pkix.Name, error) {
+	if req.TLS != nil && len(req.TLS.PeerCertificates) > 0 {
+		return &req.TLS.PeerCertificates[0].Subject, nil
```

Comment:
I would actually go all the way and put `req.TLS.PeerCertificates` itself in the context.